tests/constbuffer_ut/constbuffer_ut.c

```diff
@@ -2562,4 +2562,144 @@ TEST_FUNCTION(CONSTBUFFER_from_buffer_with_2_size_unhappy_path)
     ///clean
 }
 
+/*Tests_SRS_CONSTBUFFER_51_001: [ If `size` is 0, then CONSTBUFFER_create_writiable_handle shall fail and return NULL. ]*/
+TEST_FUNCTION(CONSTBUFFER_create_writable_handle_invalid_args_fails)
+{
+    ///arrange
+
+    ///act
+    CONSTBUFFER_WRITABLE_HANDLE handle = CONSTBUFFER_create_writable_handle(0);
+
+    ///assert
+    ASSERT_IS_NULL(handle);
+
+    ///cleanup
+}
+
+/*Tests_SRS_CONSTBUFFER_51_003: [ If any error occurs, `CONSTBUFFER_create_writiable_handle` shall fail and return NULL. ]*/
+TEST_FUNCTION(CONSTBUFFER_create_writable_handle_fails_when_malloc_fails)
+{
+    ///arrange
+    CONSTBUFFER_WRITABLE_HANDLE handle;
+    STRICT_EXPECTED_CALL(malloc_flex(IGNORED_ARG, BUFFER1_length, 1))
+        .SetReturn(NULL);
+
+    ///act
+    handle = CONSTBUFFER_create_writable_handle(BUFFER1_length);
+
+    ///assert
+    ASSERT_IS_NULL(handle);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    ///cleanup
+}
+
+/*Tests_SRS_CONSTBUFFER_51_004: [ `CONSTBUFFER_create_writiable_handle` shall set the ref count of the newly created `CONSTBUFFER_WRITABLE_HANDLE` to 1. ]*/
+TEST_FUNCTION(CONSTBUFFER_create_writiable_handle_is_ref_counted_1)
+{
+    ///arrange
+    CONSTBUFFER_WRITABLE_HANDLE handle = CONSTBUFFER_create_writable_handle(BUFFER1_length);
+    umock_c_reset_all_calls();
+    STRICT_EXPECTED_CALL(free(IGNORED_ARG));
+
+    ///act
+    CONSTBUFFER_DecRef((CONSTBUFFER_HANDLE) handle);

```

Comment:
>(CONSTBUFFER_HANDLE)  [](http://example.com/codeflow?start=23&length=21)

I know why the cast is needed but I think it is rather ugly actually. I think we should really consider having a IncRef/DecRef for the writeable one separately with the proper handles so we avoid the user doing the casts.

Reply:
Added